renovate.json5

```diff
@@ -81,6 +82,19 @@
       groupSlug: "all-npm-minor-patch",
       matchManagers: [ "npm" ],
     },
+    {
+      // major version changes should go to the next branch rather than main
+      groupName: "all major dependencies",
+      matchUpdateTypes: ["major"],
+      // Exclude all v0 dependencies from the major group. Changes to v0 packages
+      // should be treated as major updates, but renovate treats all patches and minors
+      // the same.
+      matchCurrentVersion: "!/^[\^~]?0/",
```

Comment:
Hmm...I think it would be preferable to get 0.x updates on to the next branch (since they are semver major, as the comment notes), but it might be tricky to group them into this rule.

You could create an additional rule for capturing minor/patch v0.x bumps if you drop the `!`, then use the same group name `all major dependencies` to include them.

In this rule, with the `matchUpdateTypes` as `major` only, I expect renovate to explicitly ignore v0 packages, so I don't think this is necessary at all for this rule.